packages/nx/src/config/project-graph.ts

```diff
@@ -114,15 +120,21 @@ export interface ProjectGraphProjectNode {
  *
  */
 export interface ProjectGraphExternalNode {
-  type: 'npm';
-  name: `npm:${string}`;
+  type: 'npm' | 'cargo' | 'gradle';
+  name: string;
   data: {
     version: string;
     packageName: string;
     hash?: string;
   };
 }
 
+export function isProjectGraphExternalNode(
+  node: ProjectGraphProjectNode | ProjectGraphExternalNode
+): node is ProjectGraphExternalNode {
+  return node.type === 'npm' || node.type === 'cargo' || node.type === 'gradle';
```

Comment:
If node.type is just a string to support things like `nuget` and `pip`, this check should probably be checking that the node is not a library, app, or e2e instead of a whitelist of external node types.